server/src/main/java/keywhiz/service/daos/GroupDAO.java

```diff
@@ -93,22 +118,57 @@ public void deleteGroup(Group group) {
           .set(SECRETS.OWNER, (Long) null)
           .where(SECRETS.OWNER.eq(group.getId()))
           .execute();
+      DSL.using(configuration)
+          .update(GROUPS)
+          .set(GROUPS.OWNER, (Long) null)
+          .where(GROUPS.OWNER.eq(group.getId()))
+          .execute();
     });
   }
 
   public Optional<Group> getGroup(String name) {
     GroupsRecord r = dslContext.fetchOne(GROUPS, GROUPS.NAME.eq(name));
-    return Optional.ofNullable(r).map(groupMapper::map);
+    return Optional.ofNullable(toGroup(r));
   }
 
   public Optional<Group> getGroupById(long id) {
     GroupsRecord r = dslContext.fetchOne(GROUPS, GROUPS.ID.eq(id));
-    return Optional.ofNullable(r).map(groupMapper::map);
+    return Optional.ofNullable(toGroup(r));
   }
 
   public ImmutableSet<Group> getGroups() {
-    List<Group> r = dslContext.selectFrom(GROUPS).fetch().map(groupMapper);
-    return ImmutableSet.copyOf(r);
+    Result<GroupsRecord> records = dslContext.selectFrom(GROUPS).fetch();
+
+    Map<Long, Group> groupsById = records.stream()
+        .map(groupMapper::map)
+        .collect(Collectors.toMap(Group::getId, Function.identity()));
+
+    for (GroupsRecord record : records) {
+      if (record.getOwner() != null) {
+        Group group = groupsById.get(record.getId());
+        Group owner = groupsById.get(record.getOwner());
+        group.setOwner(owner.getName());
+      }
+    }
+
+    return ImmutableSet.copyOf(groupsById.values());
+  }
+
+  private Group toGroup(GroupsRecord record) {
+    if (record == null) {
+      return null;
+    }
+
+    Group group = groupMapper.map(record);
+
+    if (record.getOwner() != null) {
+      GroupsRecord ownerRecord = dslContext.fetchOne(GROUPS, GROUPS.ID.eq(record.getOwner()));
+      if (ownerRecord != null) {
+        group.setOwner(ownerRecord.getName());
+      }
+    }
+
+    return group;
   }
```

Comment:
I believe this is a lot cleaner than the prior change for secrets, which is probably worth revisiting.